lib/devices/ios/xcode.js

```diff
@@ -29,18 +72,26 @@ xcode.getPath = function (cb) {
 
 xcode.getVersion = function (cb) {
   xcode.getPath(function (err, xcodePath) {
-    if (err  !== null) {
-      cb(err, null);
+    if (err === null && !_hasExpectedSubDir(xcodePath)) {
+      err = new Error("Expected a path ending with '" + XCODE_SUBDIR + "', but got " + xcodePath);
+    }
+    if (err !== null) {
+      return cb(err, null);
```

Comment:
since you're returning here, we don't need to put everything after this in an `else` (so we can save indentation space)